web/packages/teleport/src/services/discovery/discovery.ts

```diff
@@ -42,15 +44,40 @@ export function makeDiscoveryConfig(rawResp: DiscoveryConfig): DiscoveryConfig {
   };
 }
 
-function makeAws(rawResp: AwsMatcher[]) {
-  if (!rawResp) {
+function makeAws(rawAwsMatchers): AwsMatcher[] {
+  if (!rawAwsMatchers) {
+    return [];
+  }
+
+  return rawAwsMatchers.map(a => ({
+    types: a.types || [],
+    regions: a.regions || [],
+    tags: a.tags || {},
+    integration: a.integration,
+    kubeAppDiscovery: !!a.kube_app_discovery,
+  }));
+}
+
+function makeDiscoveryConfigReq(inputReq: DiscoveryConfig) {
+  const { name, discoveryGroup, aws } = inputReq;
+
+  return {
+    name,
+    discoveryGroup,
+    aws: makeAwsMatchersReq(aws),
+  };
+}
```

Comment:
i think this is simple enough to just inline it like this:

```
   .post(cfg.getDiscoveryConfigUrl(clusterId), {
      name: req.name,
      discoveryGroup: req.discoveryGroup,
      aws: makeAwsMatchersReq(req.aws),
    })
```

Reply:
Done.